products/revenue_analytics/backend/models.py

```diff
@@ -0,0 +1,222 @@
+    "__original_amount": IntegerDatabaseField(name="amount", hidden=True),
+    "original_amount": ExpressionField(
+        isolate_scope=True,
+        expr=ast.Call(
+            name="toDecimal",
+            args=[
+                ast.Field(chain=["__original_amount"]),
+                ast.Constant(value=EXCHANGE_RATE_DECIMAL_PRECISION),
+            ],
+        ),
+        name="original_amount",
+    ),
+    "is_refund": BooleanDatabaseField(name="refunded"),
+    "__original_currency": StringDatabaseField(name="currency", hidden=True),
+    "original_currency": ExpressionField(
+        isolate_scope=True,
+        expr=ast.Call(name="upper", args=[ast.Field(chain=["__original_currency"])]),
+        name="original_currency",
+    ),
+    # As per above, Stripe represents most currencies with integer amounts multiplied by 100,
+    # so we need to divide by 100 to get the actual amount
+    "currency_is_zero_decimal": ExpressionField(
+        isolate_scope=True,
+        expr=ast.CompareOperation(
+            op=ast.CompareOperationOp.In,
+            left=ast.Field(chain=["original_currency"]),
+            right=ast.Array(
+                exprs=[ast.Constant(value=currency.value) for currency in ZERO_DECIMAL_CURRENCIES_IN_STRIPE]
+            ),
+        ),
+        name="currency_is_zero_decimal",
+        hidden=True,
+    ),
+    "amount_decimal_divider": ExpressionField(
+        isolate_scope=True,
+        expr=ast.Call(
+            name="if",
+            args=[
+                ast.Field(chain=["currency_is_zero_decimal"]),
+                ast.Call(
+                    name="toDecimal", args=[ast.Constant(value=1), ast.Constant(value=EXCHANGE_RATE_DECIMAL_PRECISION)]
+                ),
+                ast.Call(
+                    name="toDecimal",
+                    args=[ast.Constant(value=100), ast.Constant(value=EXCHANGE_RATE_DECIMAL_PRECISION)],
+                ),
+            ],
+        ),
+        name="amount_decimal_divider",
+        hidden=True,
+    ),
+    # Amount taking into consideration whether this was a refund or not
+    # plus the decimal divider
+    "adjusted_original_amount": ExpressionField(
+        isolate_scope=True,
+        expr=ast.Call(
+            name="divideDecimal",
+            args=[
+                ast.Call(
+                    name="if",
+                    args=[
+                        ast.Field(chain=["is_refund"]),
+                        ast.Call(name="negate", args=[ast.Field(chain=["original_amount"])]),
+                        ast.Field(chain=["original_amount"]),
+                    ],
+                ),
+                ast.Field(chain=["amount_decimal_divider"]),
+            ],
+        ),
+        name="adjusted_original_amount",
+    ),
+    # Allow us to sort by timestamp properly in insights (and split in buckets)
+    "timestamp": DateTimeDatabaseField(name="created_at"),
+}
+
+
+class RevenueAnalyticsRevenueView(Table):
```

Comment:
Is this necessary to create another type? I'm wondering if we could just have these fields written in as a "SavedQuery" https://github.com/PostHog/posthog/blob/22504acef2da55e8d8bc4d2648c002b321166226/posthog/hogql/database/models.py#L231

Reply:
Yeah, reusing this was much easier, thanks for the suggestion!